app/src/main/java/com/github/dedis/student20_pop/MainActivity.java

```diff
@@ -90,25 +86,9 @@ public void onClick(View view) {
                     if (PrivateInfoStorage.storeData(this, app.getPerson().getId(), app.getPerson().getAuthentication()))
                         Log.d(TAG, "Stored private key of organizer");
 
-                    CompletableFuture.supplyAsync(() -> {
-                        try {
-                            //TODO Get URL dynamically
-                            return PoPClientEndpoint.connectToServer(URI.create("ws://10.0.2.2:2020/"), app.getPerson());
-                        } catch (DeploymentException e) {
-                            e.printStackTrace();
-                            return null;
-                        }
-                    }).thenCompose(p -> {
-                        if (p != null)
-                            return p.createLao(lao.getName(), lao.getTime(), lao.getTime(), app.getPerson().getId());
-                        else
-                            return null;
-                    }).whenComplete((errCode, t) -> {
-                        if (t != null)
-                            //TODO Show toast for error
-                            Log.e(TAG, "Error while creating the lao", t);
-                        else {
-                            //TODO Show toast for success
+                    app.getLocalProxy()
+                        .thenCompose(p -> p.createLao(lao.getName(), lao.getTime(), lao.getTime(), app.getPerson().getId()))
+                        .thenAccept(code -> {
                             Person organizer = app.getPerson().setLaos(Collections.singletonList(lao.getId()));
```

Comment:
not necessarily  related to this PR, but shouldn't we add a "addLao" method to Person and use addLao here? For the case where this particular person already has Laos.